### ISSUE TITLE
[Regression] Lightning Returns [BLUS31164]: Freezes on startup

### ISSUE THREAD

**Issue by Shoegzer** [RPCS3/rpcs3#15373]
### Quick summary

This title now freezes on startup.  Several other titles tested start up with no issues using the same build.

### Details

Initially, build 4cfdb71d3a5dc6c79ce78f0c121b1d14a7fc7763 was tested as good, and build a6fa091ab3af2621777c87ccf034533e7a253f9a was tested as bad.

A bisect was attempted but failed to build one step before completing.  However, a `git bisect visualize` shows the remaining commits, one of which is the "bad" one:
```
$ git bisect visualize
commit e126af63eda8de9c4ce064e0943d1dd35ef8aa3b (refs/bisect/bad)
Author: Eladash <18193363+elad335@users.noreply.github.com>
Date:   Wed Mar 27 10:46:45 2024 +0200

    Silence cellGemGetState

commit 750359d780f3c3ed2f7962f9ac7b7ea79426c663
Author: Eladash <18193363+elad335@users.noreply.github.com>
Date:   Wed Mar 27 10:40:25 2024 +0200

    Fixup emulation stopping

commit 8588b2b11acf390ba7bbcea33c7de8c882f6c903 (HEAD)
Author: Eladash <18193363+elad335@users.noreply.github.com>
Date:   Wed Mar 27 10:36:00 2024 +0200

    vm: Fix writer lock leak

commit 4594c912a608f9e5e8c3ed82a14ccd3f2d216c4e
Author: kd-11 <15904127+kd-11@users.noreply.github.com>
Date:   Wed Mar 27 20:20:21 2024 +0300

    rsx: Move SPIRV stuff into common code
```

### Build with regression

See above

### Attach two log files

[RPCS3_good.log](https://github.com/RPCS3/rpcs3/files/14795478/RPCS3_good.log)
[RPCS3_bad.log](https://github.com/RPCS3/rpcs3/files/14795482/RPCS3_bad.log)

### System configuration

* OS: Linux Mint 21.3 / kernel 6.5
* CPU: Intel i7-10870H
* GPU: Nvidia gtx 3070 / driver 535.161

**Comment by Shoegzer** [comment]
Update: After manually testing each of the above commits, here are the results:

e126af63eda8de9c4ce064e0943d1dd35ef8aa3b: Builds, freezes on startup as reported above (all builds made after this commit exhibit freezing on startup)
750359d780f3c3ed2f7962f9ac7b7ea79426c663: Builds, and goes ingame but performance is very poor (~10fps vs. ~30fps)
8588b2b11acf390ba7bbcea33c7de8c882f6c903: Fails to build
4594c912a608f9e5e8c3ed82a14ccd3f2d216c4e: Does not exhibit the problem and can be ignored

**Comment by elad335** [comment]
I can't reproduce, maybe send you save files?

**Comment by Shoegzer** [comment]
Thanks @elad335. I tried again using the latest commit e90b39932912cdc825f2a7e1ec5e29a6ba251704, but this time I removed all save and cache files.  It still froze on boot.  However, in my custom configuration I had resolution set to 1920x1080, and when I reset it to 1280x720, it booted just fine. Can you try with 1920x1080 too? You should be able to reproduce the issue then.

**Comment by Shoegzer** [comment]
Update: I've done some more testing with this.  The freezing in the newer builds is definitely related to Accurate PPU 128 Reservations set to "Always On", and it occurs irrespective of Driver Wake-Up Delay changes.  In testing this with other games, not all of them exhibit this behavior although a few do.  For example, setting Gran Turismo 5 to "Always On" I observed that it too froze in 5 out 5 bootup attempts.  Reverting to a "good" build before the regression as noted above and keeping that setting, it booted 5 out of 5 times with no freezing.

**Comment by Zeromus24** [comment]
When Accurate PPU 128 Reservations is enabled, the game proceeds to look for files that doesn't exist upon boot. This is what causes the crash. (#15410)

**Comment by elad335** [comment]
Does it also crash with PPU Static Interpreter PPU decoder setting? (CPU tab)

**Comment by Zeromus24** [comment]
> Does it also crash with PPU Static Interpreter PPU decoder setting? (CPU tab)

Yes.
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/14901522/RPCS3.log)

**Comment by Shoegzer** [comment]
Update: To be clear, rpcs3-v0.0.31-16252-a6fa091a from PR #15364 caused this regression.  This has also been confirmed in #15410.

**Comment by digant73** [comment]
@Shoegzer is that still an issue?

**Comment by Shoegzer** [comment]
Yes, sadly this is still an issue.  In fact, due to the regression in #15364 it seems we've lost the ability to use Accurate PPU 128 Reservations entirely, so it's been difficult to test this title and and others against this setting since then.

**Comment by digant73** [comment]
you reported https://github.com/RPCS3/rpcs3/commit/e126af63eda8de9c4ce064e0943d1dd35ef8aa3b as the origin of the crash, so attached a version rolling back that minor change. Please test it and let me know if at least it fixes the crash.

[rpcs3.zip](https://github.com/user-attachments/files/27860619/rpcs3.zip)

**Comment by Shoegzer** [comment]
Hey thanks, but I'm on linux.  I'm happy to test an appimage though if you can make one?

**Comment by digant73** [comment]
I do not have a working linux env. in this moment

**Comment by Shoegzer** [comment]
It's okay, I'm patient.  I'm confident that the specific commit I mentioned from PR #15364 broke Accurate PPU 128 Reservations though, after extensive testing and bisecting this issue before.

**Comment by digant73** [comment]
@Shoegzer please test PR #19127 and provide there the result
